driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -293,8 +324,19 @@ private PooledConnection getPooledConnectionImmediately() {
     }
 
     private MongoTimeoutException createTimeoutException(final Timeout timeout) {
-        return new MongoTimeoutException(format("Timed out after %s while waiting for a connection to server %s.",
-                                                timeout.toUserString(), serverId.getAddress()));
+        int numPinnedToCursor = this.numPinnedToCursor.get();
+        int numPinnedToTransaction = this.numPinnedToTransaction.get();
+        if (numPinnedToCursor == 0 && numPinnedToTransaction == 0) {
+            return new MongoTimeoutException(format("Timed out after %s ms while waiting for a connection to server %s.",
+                    timeout.toUserString(), serverId.getAddress()));
+        } else {
+            int numOtherInUse = pool.getInUseCount() - numPinnedToCursor - numPinnedToTransaction;
+            return new MongoTimeoutException(format("Timed out after %s ms while waiting for a connection to server %s. Details: "
```

Comment:
`Timeout.toUserString` handles the string representation completely, including the time unit. This is why the code was `Timed out after %s` and not `Timed out after %s ms`. The `ms` part on lines 330 and 334 is not needed.

Reply:
Done